From 174233abde2320c37bfd1bcb32a17c567aa0f9c6 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 16 Apr 2014 19:40:37 +0000
Subject: [PATCH 0459/1000] MANTIS 0029957 ConveyorComp: CUSS1.3 Conveyor
 (without ConveyorAEA)

ConveyorDev.idl
V1.1

added/fixed comments
interface not changed

git-svn-id: svn://localhost/SelfServiceCommon/trunk@666 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 53 +++++++++++--------
 1 file changed, 32 insertions(+), 21 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index 6a9c8621..d0bd433d 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -40,6 +40,22 @@
 module conveyordev
 {
    const string interface_version = "1.1";
+/*
+using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
+  RC indicationListener::indication(in any ind);
+    with ind: DeviceComponents::indication
+      with ind.type:   conveyordev::indicationType
+      with ind.status: conveyordev::status
+      with ind.data:   conveyordev::bagInfo                 for IND_PIECE_WEIGHT, IND_PIECE_BARCODE        
+      with ind.data:   conveyordev::characteristics         for IND_CHARACTERISTICS_INFO 
+      with ind.data:   conveyordev::extendedCharacteristic  for IND_EXTENDED_CHARACTERISTICS_INFO 
+      with ind.data:   DeviceComponents::nullData           for the others
+  RC Generic::exec(in any cmd);
+    with cmd: DeviceComponents::command
+      with cmd.type:   conveyordev::commandType
+      with cmd.data:   conveyordev::bagInfo                 for CMD_DELIVERY_NOTE_SET, CMD_PIECE_RFID_WRITE
+      with cmd.data:   DeviceComponents::nullData           for the others
+*/
 
    enum modulePosition
    {
@@ -294,38 +310,33 @@ module conveyordev
    **/
    enum indicationType
    {
-      IND_CHARACTERISTICS_INFO,              /**< Device characteristics. */
-      IND_STATUS_INFO,                       /**< Current device status. */
-      IND_PIECE_ACCEPT_DONE,                 /**< Finished CMD_PIECE_ACCEPT. */
+      IND_CHARACTERISTICS_INFO,              /**< Finished CMD_CHARACTERISTICS_QUERY */
+      IND_STATUS_INFO,                       /**< Current device status. 
+                                                  Finished CMD_STATUS_QUERY or unsolicited event */
+      IND_PIECE_ACCEPT_DONE,                 /**< Finished CMD_PIECE_ACCEPT */
       IND_PIECE_WEIGHT,                      /**< Weight of piece. */
       IND_PIECE_BARCODE,                     /**< Barcode(s) of the piece. */
-      IND_PIECE_FORWARD_DONE,                /**< Finished the CMD_PIECE_FORWARD
-                                                  command. */
-      IND_PIECE_RETURN_DONE,                 /**< Finished the CMD_PIECE_RETURN
-                                                  command. */
-      IND_PARKED_FORWARD_DONE,               /**< Finished the CMD_PARKED_FORWARD
-                                                  command. */
-      IND_PARKED_RETURN_DONE,                /**< Finished the CMD_PARKED_RETURN
-                                                  command. */
+      IND_PIECE_FORWARD_DONE,                /**< Finished CMD_PIECE_FORWARD */
+      IND_PIECE_RETURN_DONE,                 /**< Finished CMD_PIECE_RETURN */
+      IND_PARKED_FORWARD_DONE,               /**< Finished CMD_PARKED_FORWARD */
+      IND_PARKED_RETURN_DONE,                /**< Finished CMD_PARKED_RETURN  */
       IND_PIECE_DELIVER,                     /**< Piece(s) is/are about to be
                                                   delivered. */
-      IND_PIECE_REMOVE_DONE,                 /**< Finished the CMD_PIECE_REMOVE. */
-      IND_PIECE_JAM_CLEARED_DONE,            /**< Finished the CMD_PIECE_JAM_CLEARED
-                                                  command . */
+      IND_PIECE_REMOVE_DONE,                 /**< Finished CMD_PIECE_REMOVE. */
+      IND_PIECE_JAM_CLEARED_DONE,            /**< Finished CMD_PIECE_JAM_CLEARED */
       IND_PIECE_ACCEPT_WAIT,                 /**< Waiting for customer to place a
                                                   piece of baggage. */
       IND_PIECE_PARKED,                      /**< Piece is in park position. */
       IND_PIECE_ACCEPT_WORKING,              /**< Prepare weighing the piece. */
       IND_PIECE_RFID,                        /**< RFID(s) of the piece. */
-      IND_PIECE_RFID_WRITE_DONE,             /**< Finished the CMD_PIECE_RFID_WRITE
-                                                  command. */
-      IND_DELIVERY_NOTE_SET_DONE,            /**< The delivery note is set. */
+      IND_PIECE_RFID_WRITE_DONE,             /**< Finished CMD_PIECE_RFID_WRITE */
+      IND_DELIVERY_NOTE_SET_DONE,            /**< Finished CMD_DELIVERY_NOTE_SET */
       IND_PIECE_CONVEYED,                    /**< The piece is on the way. */
-      IND_PIECE_MONITOR_START_DONE,          /**< Started sending weight changes. */
-      IND_PIECE_MONITOR_STOP_DONE,           /**< Stopped sending weight changes. */
+      IND_PIECE_MONITOR_START_DONE,          /**< Finished CMD_PIECE_MONITOR_START */
+      IND_PIECE_MONITOR_STOP_DONE,           /**< Finished CMD_PIECE_MONITOR_STOP */
       IND_PARKED_FORWARD_WORKING,            /**< Delivery failed retrying. */
-      IND_RETURN_DONE,                       /**< Finished unconditional return. */
-      IND_MAINTENANCE_DONE,                  /**< Maintenance operation completed. */
+      IND_RETURN_DONE,                       /**< Finished CMD_RETURN */
+      IND_MAINTENANCE_DONE,                  /**< Finished CMD_MAINTENANCE */
       IND_EXTENDED_CHARACTERISTICS_INFO,     /**< Finished CMD_EXTENDED_CHARACTERISTICS_QUERY */
       IND_INSERTION_PROCESS_DONE,            /**< Finished CMD_INSERTION_PROCESS */
       IND_VERIFICATION_PROCESS_DONE,         /**< Finished CMD_VERIFICATION_PROCESS */
-- 
2.41.0.windows.1

